p4-16/psa/PSA.mdk

```diff
@@ -1130,36 +1130,165 @@ return whatever distribution is supported by the target?
 [INCLUDE=psa.p4:Random_extern]
 ```
 
+
 ## Action Profile
 
 Action profiles are used as table implementation attributes.
 
-Action profiles implement a mechanism to populate table entries
-with actions and action data. The only data plane operation
-required is to instantiate this extern. When the control plane adds
-entries (members) into the extern, they are essentially populating
-the corresponding table entries.
+Action profiles provide a mechanism to populate table entries
+with action specifications that have been defined outside the table entry
+specification. An action profile extern can be instantiated as
+a resource in the P4 program. A table that uses this action profile
+must specify its implementation attribute as the action profile 
+instance.
+
+An action profile instance contains member entries consisting of action
+specifications. Member entries are indexed by controller-assigned integer ids.
```

Comment:
I don't think this is relevant to dataplane behavior. This is one possible runtime API, but there are many others. Member entries can be identified by switch-assigned handles for example. I would also not mention any "controller" entity as it ties PSA to a SDN deployment.